lib/manageiq/api/client/acts_as_ar_query.rb

```diff
@@ -0,0 +1,175 @@
+module ManageIQ
+  module API
+    class Client
+      class ActsAsArQuery
+        include Enumerable
+        attr_accessor :klass, :options
+
+        SORT_ASC = "asc".freeze
+        SORT_DESC = "desc".freeze
+
+        def initialize(klass, opts = nil)
+          @klass = klass
+          @options = opts || {}
+        end
+
+        def where(*val)
```

Comment:
the ar implementation here is tricky.

in our implementation `where.not(:id => 5)` actually calls `where({}).not({:id => 5})` and the first one is a no-op